k256/src/arithmetic/mul.rs

```diff
@@ -81,26 +77,20 @@ where
 impl<P: Identity + ConditionallySelectable + Neg<Output = P>> LookupTable<P> {
     /// Given -8 <= x <= 8, returns x * p in constant time.
     fn select(&self, x: i8) -> P {
-        debug_assert!(x >= -8);
-        debug_assert!(x <= 8);
-
-        // Compute xabs = |x|
-        let xmask = x >> 7;
-        let xabs = (x + xmask) ^ xmask;
-
-        // Get an array element in constant time
-        let mut t = P::identity();
-        for j in 1..9 {
-            let c = (xabs as u8).ct_eq(&(j as u8));
-            t.conditional_assign(&self.0[j - 1], c);
-        }
-        // Now t == |x| * p.
+        debug_assert!((-8..=8).contains(&x));
```

Comment:
I'm not sure what this does, does it not test all 16 values? I think the old test is fine